pkg/apis/azure/validation/secrets.go

```diff
@@ -34,17 +34,33 @@ var (
 func ValidateCloudProviderSecret(secret, oldSecret *corev1.Secret) error {
 	secretKey := fmt.Sprintf("%s/%s", secret.Namespace, secret.Name)
 
-	for _, key := range []string{azure.SubscriptionIDKey, azure.TenantIDKey, azure.ClientIDKey, azure.ClientSecretKey} {
+	for _, key := range []string{azure.SubscriptionIDKey, azure.TenantIDKey} {
 		val, ok := secret.Data[key]
 		if !ok {
 			return fmt.Errorf("missing %q field in secret %s", key, secretKey)
 		}
 		if len(val) == 0 {
 			return fmt.Errorf("field %q in secret %s cannot be empty", key, secretKey)
 		}
+
+		// subscriptionID and tenantID must be valid GUIDs,
+		// see https://docs.microsoft.com/en-us/rest/api/securitycenter/locations/get
+		if !guidRegex.Match(val) {
+			return fmt.Errorf("field %q in secret %s must be a valid GUID", key, secretKey)
+		}
+	}
+
+	for _, key := range []string{azure.ClientIDKey, azure.ClientSecretKey} {
```

Comment:
Maybe I misunderstand the check, but AFAIK if ClientIDKey or ClientSecretKey are missing then both must be missing right ?

If so, then this check is too loose as it only checks if they are okay individually and not as a set of properties that exist (or not) together.

Reply:
Make completely sense. I adapted the validator to check if both is set or none.